cli/app.go

```diff
@@ -2748,6 +2748,55 @@ This won't work unless you have an existing installation of our GitHub app on yo
 				},
 			},
 		},
+		{
+			Name:            "inference",
+			Usage:           "work with cloud hosted inference service",
+			UsageText:       createUsageText("inference", nil, false, true),
+			HideHelpCommand: true,
+			Subcommands: []*cli.Command{
+				{
+					Name:  "infer",
+					Usage: "run inference on an image",
+					UsageText: createUsageText("inference infer", []string{
+						generalFlagOrgID, inferenceFlagFileOrgID, inferenceFlagFileID,
+						inferenceFlagFileLocationID, inferenceFlagModelID, inferenceFlagModelVersionID,
+					}, true, false),
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     generalFlagOrgID,
+							Usage:    "organization ID that is executing the inference job",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     inferenceFlagFileOrgID,
+							Usage:    "organization ID that owns the file to run inference on",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     inferenceFlagFileID,
+							Usage:    "file ID of the file to run inference on",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     inferenceFlagFileLocationID,
+							Usage:    "location ID of the file to run inference on",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     inferenceFlagModelID,
```

Comment:
I think we should try to have parity with the training CLI, where we specify model-org-ID and model-name to uniquely identify the registry ID https://github.com/viamrobotics/rdk/blob/main/cli/ml_training.go#L22

I think we should either have both training and inference require:
- model org ID + name + version OR
- registry ID + version

I'll leave it up to you decide what makes the most sense.